### PR TITLE
move redaction to function that takes place after collection

### PR DESCRIPTION
**Pull request by gravitypriest**
Signed-off-by: Jeremy Crafts <jcrafts@redhat.com>

### CODE REVIEW

File: insights/client/data_collector.py [hunk 1]
```diff
@@ -257,6 +250,65 @@ def run_collection(self, conf, rm_conf, branch_info, blacklist_report):
         self._write_blacklist_report(blacklist_report)
         logger.debug('Metadata collection finished.')
 
+    def data_redaction(self, rm_conf):
```
Comment (jhjaggars) [inline insights/client/data_collector.py:253]: Maybe call this `redact`?

File: insights/client/data_collector.py [hunk 2]
```diff
@@ -257,6 +250,65 @@ def run_collection(self, conf, rm_conf, branch_info, blacklist_report):
...
+
+        for dirpath, dirnames, filenames in os.walk(self.archive.archive_dir):
+
+            for f in filenames:
+                fullpath = os.path.join(dirpath, f)
+
+                logger.debug('Processing %s...', fullpath)
+
+                # password removal
+                sedcmd = Popen(['sed', '-rf', constants.default_sed_file, fullpath], stdout=PIPE)
```
Comment (jhjaggars) [inline insights/client/data_collector.py:294]: moving this logic to its own function would make it easier to test in isolation.

File: insights/client/insights_spec.py [hunk 3]
```diff
@@ -18,32 +18,17 @@ class InsightsSpec(object):
...
 
 
 class InsightsCommand(InsightsSpec):
     '''
     A command spec
     '''
-    def __init__(self, config, spec, exclude, mountpoint):
-        InsightsSpec.__init__(self, config, spec, exclude)
+    def __init__(self, config, spec, mountpoint):
+        InsightsSpec.__init__(self, config, spec)
```
Comment (jhjaggars) [inline insights/client/insights_spec.py:31]: Any idea why this isn't using `super`?

Comment (gravitypriest) [reply insights/client/insights_spec.py:31]: Written many moons ago when I was younger and not as smart. I'll update it since I'm cleaning house anyway

**Comment by gravitypriest** [comment]
@jhjaggars Updated

**Pull request merged**